### DISCUSSION TITLE
Best Way To Debug Slow Responses

### DISCUSSION THREAD

**Discussion by illusive-ch** [nuwave/lighthouse#2587] [Q&A]
What is the best way to debug a slow response from a graphql request.
```graphql
mutation Login {
    login(
        input: {
            email: "test@test.com"
            password: "password"
            brand_id: "01hpce7egdw8y4fb4nav1zx7sz"
        }
    ) {
        token
        user {
            id
            email_verified_at
        }
    }
}
```
This is taking 1.8-2.2 seconds to process on the server but can't figure out why.

**Comment by spawnia** [comment]
I have had some success with https://github.com/itsgoingd/clockwork.

**Comment by illusive-ch** [comment]
@spawnia this worked nicely, for some reason I could not get telescope to work on my vapor staging enviroment but clockwork works  80% of the time. ty For anyone using algolia make sure that you have your queue set to true :)

**Comment by illusive-ch** [comment]
@spawnia have you ever tried to get telescope to play nicely with lighthouse? Its not showing up in any of the requests. Clockwork is a bit hard to drill into specific issues and seems very sporadic what/when it can collect.

**Reply by spawnia** [reply]
No.

**Comment by stayallive** [comment]
This is slightly off-topic and a little late maybe but a login is usually slower then usual because of password hashing. The best way to debug this is by using a profiler like XDebug, Blackfire or in production by using Tideways or Sentry.
